File: TP3/JAVA/src/test/resources/TestStrategy.java
package test.resources;

import org.junit.jupiter.api.Test;

import main.resources.ITriStrategy;
import src.Cours;
import src.Etudiant;

import static org.junit.jupiter.api.Assertions.*;

import java.util.List;

public class TestStrategy {
    
    @Test
    void testTriParNom() {
        Cours c1 = Cours("Mathématiques", "M.Dupond");
        Etudiant e1 = Etudiant("Zoe", 20, "E001", c1);
        Etudiant e2 = Etudiant("Alice", 22, "E002", c1);
        
        ITriStrategy tri = TriParNom();
        List<Etudiant> resultat = tri.trier(List.of(e1, e2));
        
        assertEquals(resultat.get(0).get_nom(), "Alice");
        assertEquals(resultat.get(1).get_nom(), "Zoe");
    }

    @Test
    void testTriParMoyenne() {
        Cours c1 = Cours("Mathématiques", "M.Dupond");
        Etudiant e1 = Etudiant("Zoe", 20, "E001", c1);
        Etudiant e2 = Etudiant("Alice", 22, "E002", c1);
        e1.ajouterNote(10);
        e2.ajouterNote(20);
        
        ITriStrategy tri = TriParNom();
        List<Etudiant> resultat = tri.trier(List.of(e1, e2));
        
        assertEquals(resultat.get(0).get_nom(), "Alice");
        assertEquals(resultat.get(0).get_moyenne(), 20);
    }
}

File: TP3/PYTHON/test/test_strategy.py
from src.codes_initiaux.etudiant import Etudiant
from src.pattern.Strategy.TriParNom import TriParNom
from src.pattern.Strategy.TriParMoyenne import TriParMoyenne
from src.codes_initiaux.cours import Cours

def test_tri_par_nom():
    c1 = Cours("Mathématiques", "M.Dupond")
    e1 = Etudiant("Zoe", 20, "E001", [c1])
    e2 = Etudiant("Alice", 22, "E002", [c1])
    
    tri = TriParNom()
    resultat = tri.trier([e1, e2])
    
    assert resultat[0].get_nom() == "Alice"
    assert resultat[1].get_nom() == "Zoe"

def test_tri_par_moyenne():
    c1 = Cours("Mathématiques", "M.Dupond")
    e1 = Etudiant("Bob", 20, "E001", [c1])
    e2 = Etudiant("Alice", 22, "E002", [c1])
    e1.ajouter_note(10) # Moyenne 10
    e2.ajouter_note(20) # Moyenne 20
    
    tri = TriParMoyenne()
    resultat = tri.trier([e1, e2])
    
    assert resultat[0].get_moyenne() == 20.0
    assert resultat[0].get_nom() == "Alice"